conans/test/functional/editable/consume_header_only_test.py

```diff
@@ -0,0 +1,164 @@
+# coding=utf-8
+
+import os
+import tempfile
+import unittest
+import textwrap
+
+from parameterized import parameterized
+
+from conans.model.ref import ConanFileReference
+from conans.paths import LINKED_FOLDER_SENTINEL
+from conans.paths.package_layouts.package_editable_layout import CONAN_PACKAGE_LAYOUT_FILE
+from conans.test import CONAN_TEST_FOLDER
+from conans.test.utils.tools import TestClient
+from conans.util.files import save
+
+
+class HeaderOnlyLibTestClient(TestClient):
```

Comment:
For me, extending the ``TestClient`` to implement the saving of some files seems a bit complex to read. I would prefer to pass the client as argument to functions that do what you want on such a client. I am ok with this as-is, just my opinion, I would prefer it the other way, but no need to change it.

Reply:
I agree